feat: add web search tool support and enhance content parsing

ncoop57 opened the pull request:
This PR adds an example for the Claude web search tool and also updates the contents function to better support multiple text blocks and responses and text blocks that contain citations. While this update to the contents function was made specifically in response to the web search tool update, I believe that this format will be also very useful for the citations API more broadly for uploaded documents.

### Changes

- Added example for Claude's web search tool with function to easily configure tool parameters
- Enhanced content parsing to handle citations and multiple content blocks
- Added reference list formatting for cited sources using markdown format

jph00 commented:
Thanks for this contribution.

Can you please revise it -- the quality of the code and resulting docs it well under the quality of the rest of this nb tbh. We want each contribution to our libs to *improve* the quality of it. Currently this PR is just a big dump of code! Please try to make it beautiful and great.

Let me know if you'd like my help with achieving that.

jph00 commented:
Also don't forget about this PR - https://github.com/AnswerDotAI/claudette/pull/58 . If we're replacing it, we should let them know.

ncoop57 commented:
@jph00 thanks for pushing me to make it the best that I can. I've spent a significant amount of effort to improve the docs on it to be aligned with the existing ones. I've also fixed a few bugs in how we do markdown rendering for the chat and client objects.

It doesn't fully replace #58 since the web search result citations are different from the document citations added in that PR. However, I will open a follow up one that reuses the system I've added here, which will supersede it and I'll make sure to notify the author of that PR and give them credit for the code that I plan on copying over from it.

jph00 commented:
Thank you, much better! :)
